onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -247,6 +275,51 @@ Status PadCpuImpl<float>(OpKernelContext* ctx,
 
 template <>
 Status Pad<float>::Compute(OpKernelContext* ctx) const {
-  return PadCpuImpl<float>(ctx, pads_, slices_, mode_, value_);
+  // kOnnxDomain Pad opset >= 11 (Or) kMsDomain opset == 1
+  if (is_dynamic_) {
+    const Tensor& input_tensor = *ctx->Input<Tensor>(0);
+    std::vector<int64_t> output_dims(input_tensor.Shape().GetDims());
+    size_t dimension_count = output_dims.size();
+
+    const Tensor& pads_tensor = *ctx->Input<Tensor>(1);
+    const std::vector<int64_t>& pads_tensor_dims = pads_tensor.Shape().GetDims();
+    ORT_ENFORCE(pads_tensor.DataType() == DataTypeImpl::GetType<int64_t>(),
+                "Pads tensor should be an INT64 tensor");
+    ORT_ENFORCE(pads_tensor_dims.size() == 1 || (pads_tensor_dims.size() == 2 && pads_tensor_dims[0] == 1),
+                "Pads tensor should be a 1D tensor of shape [2 * input_rank] or a 2D tensor of shape [1, 2 * input_rank]");
+
+    std::vector<int64_t> pads(2 * dimension_count, 0);
+    const int64_t* pads_tensor_raw_data = pads_tensor.template Data<int64_t>();
```

Comment:
Side note:
I don't think it's significant here, but given you explicitly set every value it would be slightly more efficient to call 'reserve' instead of constructing with a default value being set for each element. In this case you'd most likely save a memset call by using the default ctor and a reserve call.

Reply:
Will fix.

Reply:
Fixed. Fixed for the `slices` vector too (few lines below).